warp10/src/main/java/io/warp10/continuum/gts/GTSEncoder.java

```diff
@@ -702,6 +717,7 @@ public synchronized int addValue(long timestamp, long location, long elevation,
             this.stream.write(buf, 0, 8);
             // Clear the last BDValue otherwise we might incorrectly encode the next value specified as a BigDecimal
             lastBDValue = null;
+            validLastDoubleValue = true;
```

Comment:
I think setting `validLastBDValue` to `false` is clearer than setting `lastBDValue` to `null`. This change goes with the first comment.